### DISCUSSION TITLE
Sonoff power switches losing mqtt messages

### DISCUSSION THREAD

**Discussion by carygravel** [arendst/Tasmota#20071] [Q&A]
I have  a several Sonoff power switches running tasmota which I have been using to control the ventilation (open/close windows, turn fan/dehumidifier on and off) in my cellar for the last couple of years all with no problem, via a Node-Red flow.

In those days, it was pulling the temperature/humidity via HTTP, and switching the Sonoffs via MQTT.

After the Raspberry Pi it was running on didn't survive an upgrade, I took the opportunity to rewrite the Node-Red flow in Python to reduce the number of dependencies and overhead and similarly, started using MQTT to pull in the temperature/humidity sensor data.

This works - sometimes, but tasmota seems to be sporadically ignoring messages.

Adding sleep() in places seems to help a bit, which makes me think that tasmota is struggling with the number of messages - but we aren't talking about more than an average of around 1 per second.

I've tried playing with things like qos without any joy.

Of course, pull the sensor data via http to reduce the traffic, but it seems to me that what I want should work.

Can anyone offer any suggestions on how to solve the problem?

**Comment by sfromis** [comment]
As you are not posting details, no basis for guesses, but if it is about having energy monitoring data up to date, frequent polling is not a good approach. Tasmota has the `PowerDelta` command/feature for the purpose, where Tasmota will generate reporting when the energy consumption level steps up or down.

**Comment by barbudor** [comment]
Why do you have to pull anyhting ?
With Teleperiod, Tasmota publishes automatically and periodically published the sensor values and your regulation process just have to receive it without any effort

**Comment by carygravel** [comment]
Apologies for the late reply. Real life and a broken phone conspired against me.

> As you are not posting details, no basis for guesses, but if it is about having energy monitoring data up to date, frequent polling is not a good approach. Tasmota has the PowerDelta command/feature for the purpose, where Tasmota will generate reporting when the energy consumption level steps up or down.

I'm using [these sensors](https://shop.openenergymonitor.com/emonth-temperature-humidity-node/) to measure temp/rel humidity with 3 nodes and [emonhub](https://github.com/openenergymonitor/emonhub) to publish the values around once a minute per node to mqtt.

So far so good. I've got some python code which subscribes to the updates, calculates the absolute humidity, and then, also using mqtt, switches two Tasmota node (one 1ch for the fan, one 2ch to open/close the window).

Because of the way emonhub publishes the messages, there are 6 messages per update per node, i.e. for 3 nodes on average 1 every 3s, and sometimes the Tasmota nodes miss the messages telling them to switch.

> Why do you have to pull anyhting ?
> With Teleperiod, Tasmota publishes automatically and periodically published the sensor values and your regulation process just have to receive it without any effort

I am not using Tasmota to sense the temp/humidity, just switch the motors.

I guess the answer is going to be to poll the sensor data using REST, and I guess that I could use REST/web requests to switch the motors, too, but it annoys me that I couldn't get it working flawlessly with mqtt.

**Reply by barbudor** [reply]
deaf dialog

We are still not understanding what you are trying to do
With a lack of detalled information, configuration and logs, this is going nowhere

**Reply by sfromis** [reply]
You can build Tasmota with an optional feature to subscribe to MQTT payloads (if that is what you are not telling us you want to do), but this may not work well if those payloads are not in a comparable format to what Tasmota is using. Much better options if running Tasmota on an ESP32-based device.
